File: ecommerce-frontend/src/types/address.ts
export interface UserAddress {
  id: number;
  province: string;
  city: string;
  detail: string;
  recipient_name: string | null;
  recipient_phone: string | null;
  is_default: boolean;
  created_at: string;
}

export interface AddressCreate {
  province: string;
  city: string;
  detail: string;
  recipient_name?: string;
  recipient_phone?: string;
  is_default?: boolean;
}

File: ecommerce-backend/app/models/address.py
# app/models/address.py
from sqlalchemy import String, Integer, Boolean, DateTime, ForeignKey, func
from sqlalchemy.orm import Mapped, mapped_column, relationship
from datetime import datetime
from app.core.database import Base


class UserAddress(Base):
    __tablename__ = "user_addresses"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    province: Mapped[str] = mapped_column(String(50), nullable=False, comment="省")
    city: Mapped[str] = mapped_column(String(50), nullable=False, comment="市")
    detail: Mapped[str] = mapped_column(String(500), nullable=False, comment="详细地址")
    recipient_name: Mapped[str | None] = mapped_column(String(50), nullable=True, comment="收件人")
    recipient_phone: Mapped[str | None] = mapped_column(String(20), nullable=True, comment="收件人电话")
    is_default: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

    user: Mapped["User"] = relationship("User", back_populates="addresses")
